### DISCUSSION TITLE
Question about lazer framerate

### DISCUSSION THREAD

**Discussion by Dear-You** [ppy/osu#17628] [Q&A]
my notebook is
i5 8400+1060 6G+ 16G
I haven't opened the stable version and it is compatible with 400. After opening it, 600+
I went to youtube to watch the recommended video of lazer and went to play with lazer.
The framerate is only 200-300 fps, and my tablet has input lag.
I'm not very good at English, so I use translation software.
I really like this game please tell me the solution

**Comment by bdach** [comment]
stable works differently from lazer in that input processing is not done on the same thread as drawing to the screen. so while you get "200 fps", if you check the frame graph, the input thread should be running at 1000fps unless you are using single threaded mode.

**Reply by Dear-You** [reply]
Thank you for your reply.
What about the delay of my digital board? My digital board is driven with an open tablet driver, and I have also used the driver of wocom. There is an obvious feeling of pointer chasing, especially in the game.

**Reply by bdach** [reply]
lazer comes with its own tablet handler (based on opentabletdriver) too. if you have several drivers active at a time then that could certainly cause problems. please try turning off any/all of them, or toggling the tablet handling off in lazer game settings.

**Reply by Dear-You** [reply]
I Have Tried To Turn Off The Driver Outside With Lazer, Or Turn Off Lazer With The Outside. There is no way to change my input delay.

**Reply by Dear-You** [reply]
And there is a strange thing.
The number of frames in my laptop playing the stable version is only 400+. There are 600+ in stable mode, but if I link to the external screen with hdmi, it can reach more than 1,000 frames. I didn't find this problem in any other games.

**Reply by bdach** [reply]
i can't help you sorry. it's likely something to do with your specific hardware configuration and you haven't provided enough info to go on (not even a video). we cannot provide tech support advice here.

**Reply by Dear-You** [reply]
It doesn't matter. Thank you for your comments. I'll study it myself.

**Reply by peppy** [reply]
it sounds like lazer is not running exclusive fullscreen. this is basically always the case for people feeling more latency. it's not framerate, it's fullscreen related.

make sure you are running in fullscreen. if you are on some intel or nvidia optimus graphics boards, you may need to wait for fixes from our end.

**Reply by Dear-You** [reply]
So that's it. I understand. So I played with the stable with compatible mode before fixing the problem.

**Reply by peppy** [reply]
Ah yep. We will be adding back a similar compatibility mode in the near future, so watch out for that.

**Reply by Dear-You** [reply]
![image](https://user-images.githubusercontent.com/93472816/162980908-872fe7e4-ac9e-409a-916d-94c413de546b.jpeg)![image](https://user-images.githubusercontent.com/93472816/162980976-de12f96d-b40c-4b9a-8a9e-48c377c14166.jpeg)![image](https://user-images.githubusercontent.com/93472816/162981050-8023fd24-1783-456a-8e07-660bd50a2453.jpeg)![image](https://user-images.githubusercontent.com/93472816/162981149-b2eb48aa-a330-4508-8ce0-40ffcecfa1f9.jpeg)

Hi! I'm back again. I went to my college roommate to borrow an external display screen to show my problem. I linked the external display to the hdmi of the laptop graphics card, and set the external display as the home screen. The frame number is more than 1,000, and the previous movement delay is gone. But if you link the CPU screen with my own notebook, it will only have more than 300 frames. Both the stable version and the lazer have this problem. This problem has a great impact on players who play osu on laptops. Many notebook players in my Chinese forum have this problem. Can you fix it?

**Reply by peppy** [reply]
Both of the numbers here are fine. 300fps is fine and already higher than you should require.

Any "delay" you are feeling is likely due to exclusive fullscreen not triggering. We are aware of some cases where it doesn't activate, and will look to fix those.

**Reply by Dear-You** [reply]
300 is good compared with other games, but the higher the frame number of music games such as osu, the lower the latency, the better you can play, especially the input delay and response speed. And the gap between 1000 and 300 is very big.

**Reply by peppy** [reply]
It's actually not. Trust me. Things will improve once exclusive fullscreen is correctly implemented.

**Reply by Dear-You** [reply]
OK, I'll look forward to it.
